classes/migration/install/AffiliationsMigration.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * @file classes/migration/install/AffiliationsMigration.php
+ *
+ * Copyright (c) 2014-2022 Simon Fraser University
+ * Copyright (c) 2000-2022 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class AffiliationsMigration
+ *
+ * @brief Describe database table structures.
+ */
+
+namespace PKP\migration\install;
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+use PKP\migration\Migration;
+
+class AffiliationsMigration extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('author_affiliations', function (Blueprint $table) {
+            $table->comment('Author affiliations');
+            $table->bigInteger('author_affiliation_id')->autoIncrement();
+            $table->bigInteger('author_id');
+            $table->string('ror')->nullable();
+
+            $table->index(['ror'], 'author_affiliations_ror');
+            $table->foreign('author_id')->references('author_id')->on('authors')->cascadeOnDelete();
+        });
+
+        Schema::create('author_affiliation_settings', function (Blueprint $table) {
+            $table->comment('More data about author affiliations');
+            $table->bigInteger('author_affiliation_setting_id')->autoIncrement();
+            $table->bigInteger('author_affiliation_id');
+            $table->string('locale', 28)->default('');
+            $table->string('setting_name', 255);
+            $table->mediumText('setting_value')->nullable();
```

Comment:
Yes, the value for name can be null. I add all current locales to the table, even if they are null.

Reply:
OK.
Note that you might need to get the possible submission locales differently, i.e. `$submission->getPublicationLanguages($context->getSupportedSubmissionMetadataLocales());`

Reply:
Hi @bozana,
I can't figure out how to get $submission in the classes PKP\core\PKPApplication\FieldAffiliations. I do see this in PKP\components\forms\publication\ContributorForm, but it's always null. 

I do see different locales in PKPApplication::get()->getRequest()->getContext(): 
- supportedAddedSubmissionLocales
- supportedDefaultSubmissionLocale
- supportedFormLocales
- supportedLocales
- supportedSubmissionLocales
- supportedSubmissionMetadataLocales

But you pointed out that that is not correct. Thanks.

Reply:
Hi @GaziYucel, you need submission only because of allowed/supported locales, right?
If so, maybe: the ContributorForm actually contains all needed information about locales, so maybe to provide them (locales that you need for the affiliation field) here https://github.com/pkp/ui-library/blob/main/src/components/ListPanel/contributors/ContributorsListPanel.vue#L393, similar as you provide the authorId?

Reply:
I already saw it there, but the variables are called different there, not sure if these have the values you wrote above. These are the locales in ContributorsListPanel:
- primaryLocale, e.g. 'en'
- supportedFormLocales, e.g. [ { key: 'en', label: 'English' }, { key: 'fr_CA', label: 'French (Canada)' } ]
- visibleLocales, e.g. [ 'en' ]
​
And the one I use now in PHP is $submissionContext->getSupportedSubmissionLocales()

Which one is the correct one than?

Reply:
The locales for the ContributorForm are provided here: https://github.com/pkp/pkp-lib/blob/main/classes/components/listPanels/ContributorsListPanel.php#L120. Those are the locales of the ContributorsListPanel.
And locales provided to the ContributorsListPanel (s. different places/handler in the code) are:
```
$locales = collect($submissionContext->getSupportedSubmissionMetadataLocaleNames() + $submission->getPublicationLanguageNames())
            ->map(fn (string $name, string $locale) => ['key' => $locale, 'label' => $name])
            ->sortBy('key')
            ->values()
            ->toArray();
```
Those are the supportedFormLocales.
The primary and visibleLocales are the same for each Form.
So I think you should only work with these locales for affiliations as well.
 
`$submissionContext->getSupportedSubmissionLocales() ` is not enough because they contain only currently enabled submission locales (and not metadata locales). The submission locales are those a submission can be published in, i.e. the article full text; and metadata locales can contain/allow additional locales for metadata -- e.g. an article in German, but the metadata also in English. 
Additionally, if already some metadata exists for a submission in a locale that is in the meantime disabled in journal settings -- lets say the metadata were entered in Finish, but then the Finish is disabled -- the users/editor still needs to be able to edit those existing metadata (in Finish). That is why we consider also $submission->getPublicationLanguageNames() (that returns all existing publication languages).

I have not taken a look into the ui-library, but will take a look where you need to locales and how...
But I believe those 3 kinds of form locales should be enough and correct for affiliations...